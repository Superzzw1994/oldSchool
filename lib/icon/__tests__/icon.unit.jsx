import * as renderer from 'react-test-renderer';
import { mount } from 'enzyme'
import React from 'react'
import Icon from '../icon'

describe('icon', () => {
  it('Icon组件快照测试', () => {
    const json = renderer.create(<Icon/>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('测试点击事件', () => {
    const fn = jest.fn()
    const component = mount(<Icon name='zzw' onClick={fn}></Icon>)
    component.find('svg').simulate('click')
    expect(fn).toBeCalled()
  })
})