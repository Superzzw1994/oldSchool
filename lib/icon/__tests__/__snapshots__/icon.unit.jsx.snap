// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon Icon组件快照测试 1`] = `
<svg
  className="oldschool_icon"
>
  <use
    xlinkHref="#undefined"
  />
</svg>
`;
